antarest/study/storage/rawstudy/ini_reader.py

```diff
@@ -0,0 +1,206 @@
+          This case is required to parse Xpansion `user/expansion/settings.ini` files
+          (using `SimpleKeyValueReader` subclass).
+
+        - If the file has duplicate sections, then the values are merged.
+          This case is required when the end-user produced an ill-formed `.ini` file.
+          This ensures the parsing is robust even if some values may be lost.
+
+        - If a section has duplicate keys, then the values are merged.
+          This case is required, for instance, to parse `settings/generaldata.ini` files which
+          has duplicate keys like "playlist_year_weight", "playlist_year +", "playlist_year -",
+          "select_var -", "select_var +", in the `[playlist]` section.
+          In this case, duplicate keys must be declared in the `special_keys` argument,
+          to parse them as list.
+
+        - If a section has no key, then an empty dictionary is returned.
+          This case is required to parse `input/hydro/prepro/correlation.ini` files.
+
+        - If a section name has square brackets, then they are preserved.
+          This case is required to parse `input/hydro/allocation/{area-id}.ini` files.
+
+        Args:
+            ini_file: file or file-like object.
+
+        Returns:
+            Dictionary of parsed `.ini` file which can be converted to JSON.
+        """
+        # NOTE: This algorithm is 1.93x faster than configparser.ConfigParser
+        sections: t.Dict[str, t.Dict[str, t.Any]] = {}
+        section_name = self._section_name
+
+        for line in ini_file:
+            line = line.strip()
+            if not line or line.startswith(";") or line.startswith("#"):
+                continue
+            elif line.startswith("["):
+                section_name = line[1:-1]
+                sections.setdefault(section_name, {})
+            elif "=" in line:
+                key, value = map(str.strip, line.split("=", 1))
+                section = sections.setdefault(section_name, {})
+                if key in self._special_keys:
+                    section.setdefault(key, []).append(value)
+                else:
+                    section[key] = value
+            else:
+                raise ValueError(f"☠☠☠ Invalid line: {line!r}")
+
+        return sections
+
+
+class SimpleKeyValueReader(IniReader):
+    """
+    Simple INI reader for "settings.ini" file which has no section.
+    """
+
+    def read(self, path: t.Any) -> JSON:
+        """
+        Parse `.ini` file which has no section to JSON object.
+
+        This class is required to parse Xpansion `user/expansion/settings.ini` files.
+
+        Args:
+            path: Path to `.ini` file or file-like object.
+
+        Returns:
+            Dictionary of parsed key/value pairs.
+        """
+        sections = super().read(path)
+        obj = t.cast(t.Mapping[str, JSON], sections)
+        return obj[self._section_name]
+
+
+class MultipleSameKeysIniReader(IniReader):
```

Comment:
This class has no methods, or attributes. Could you explain why wouldn't you directly use IniReader.

Reply:
Je voulais éviter de toucher à trop de fichiers en faisant ce refactoring. Mais tu as raison, cela ne fait pas de sens et je vais corriger cela.